rust/src/delta.rs

```diff
@@ -371,14 +380,25 @@ pub struct DeltaTableState {
     min_reader_version: i32,
     min_writer_version: i32,
     current_metadata: Option<DeltaTableMetaData>,
+    tombstone_retention_timestamp: DeltaDataTypeTimestamp,
 }
 
 impl DeltaTableState {
     /// Full list of tombstones (remove actions) representing files removed from table state).
-    pub fn tombstones(&self) -> &Vec<action::Remove> {
+    pub fn all_tombstones(&self) -> &Vec<action::Remove> {
         self.tombstones.as_ref()
     }
 
+    /// List of non-expired tombstones (remove actions) representing files removed from table state.
+    /// The retention period is set by `deletedFileRetentionDuration` with default value of 1 week.
+    pub fn tombstones(&self) -> Vec<action::Remove> {
+        self.tombstones
+            .iter()
+            .filter(|t| t.deletion_timestamp > self.tombstone_retention_timestamp)
+            .map(|t| t.clone())
+            .collect()
+    }
```

Comment:
Need help on naming things. 

In order to not break things (for current vaccum func), I'm leaving the function to retrieve all tombstones, but renaming it to  `all_tombstones`, whereas `tombstones` will return only active tombstones. This is the default behavior I found in spark delta.

wdyt? Or we should just leave `tombstones` instead and create new `active_tombstones` / `unexpired_tombstones`?

Reply:
imho for naming - i like exactly two functions with specific names -  `all_tombstones` (aka the entire set) and `unexpired_tombstones` (aka the subset of unexpired).